### PR TITLE
Added OSX module

### PR DESCRIPTION
**Pull request by CodeMaxx**
The OSX module works now. I have added one script as of now which is for `execve` function.

@Ali-Razmjoo @jowasp

### CODE REVIEW

File: core/commands.py [hunk 1]
```diff
@@ -46,6 +46,10 @@
 ['none']},
 
 					},
+				'osx_x86' :  #generate sub command - os name
+					{
+						'exec' : {'file_to_execute':['none','xor_random','xor_yourvalue','add_random','add_yourvalue','sub_random','sub_yourvalue','inc','inc_timesyouwant','dec','dec_timesyouwant','mix_all']}, #function of shellcode
```
Comment (Ali-Razmjoo) [inline core/commands.py]: Please remove `'xor_random','xor_yourvalue','add_random','add_yourvalue','sub_random','sub_yourvalue','inc','inc_timesyouwant','dec','dec_timesyouwant','mix_all'` encodes and just leave the `none`.

Comment (CodeMaxx) [inline core/commands.py]: Just did

**Comment by Ali-Razmjoo** [comment]
hello, I think we need to change something in shellcode to bypass the protection before add it in zsc,
```
Johannas-iMac:owasp-zsc-osx johannacuriel$ gcc z-bin-sh.c -o z-bin-sh
Johannas-iMac:owasp-zsc-osx johannacuriel$ ./z-bin-sh
Illegal instruction: 4
```

**Comment by jowasp** [comment]
@CodeMaxx this shellcode of this file executes fine
```
#include <stdio.h>
#include <sys/mman.h>
#include <string.h>
#include <stdlib.h>
  
int (*sc)();
  
char shellcode[] =
"\x68\x90\x90\x90\x6d\x5b\xc1\xeb\x10\xc1\xeb\x08\x53\x68\x78\x2e\x61\x73\x68\x6d\x5f\x6f\x73\x68\x6f\x5f\x61\x73\x68\x68\x65\x6c\x6c\x89\xe3\x31\xc0\x50\x89\xe2\x53\x89\xe1\x52\x51\x53\x31\xc0\xb0\x3b\x6a\x2a\xcd\x80\xb0\x01\xb3\x01\xcd\x80";
  
int main(int argc, char **argv) {
  
    void *ptr = mmap(0, 0x22, PROT_EXEC | PROT_WRITE | PROT_READ, MAP_ANON
            | MAP_PRIVATE, -1, 0);
  
    if (ptr == MAP_FAILED) {
        perror("mmap");
        exit(-1);
    }
  
    memcpy(ptr, shellcode, sizeof(shellcode));
    sc = ptr;
  
    sc();
  
    return 0;
}
```

the one generate by ZSC has issues

**Comment by CodeMaxx** [comment]
Refer to this discussion here

https://groups.google.com/forum/#!topic/owasp-zsc/3MftTqbFRiI

**Comment by CodeMaxx** [comment]
Added `system()` as well. I have used 2-3 lines from `run` function in `system.py` for linux. I modified it make it work for all kinds of input. I didn't pay attention to what bug it is talking about. But as of now its working perfectly.

**Comment by CodeMaxx** [comment]
I think now we have good amount of stuff to show at DEFCON. I'll see what more I can add in the coming week. My college is starting so I won't have a lot of time. I'll try to help in the presentation as well.

Cheers!

**Comment by Ali-Razmjoo** [comment]
hello, thanks for contributing.

**Pull request merged**